### PR TITLE
Interactive viewer doesn't appear to respect the trackpadScrollCausesScale parameter

### PR DESCRIPTION
**Pull request by JsGjKJzi**
PR https://github.com/flutter/flutter/pull/114280 does not actually restore the functionality to pre https://github.com/flutter/flutter/pull/114280 behavior, where a two-finger scroll up/down on a trackpad results in a scale (zoom) even when the trackpadScrollCausesScale parameter is set to `true`.

This PR causes InteractiveViewer to behave as expected and fixes issue https://github.com/flutter/flutter/issues/127111.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by moffatman** [comment]
Fix looks correct, but will need a test in

https://github.com/flutter/flutter/blob/2d6c67f4a4b73ce406587d8a06755d31ce5fc97b/packages/flutter/test/widgets/interactive_viewer_test.dart#L1807

Probably just a few lines, adding a use of `pointer.scroll(center, offset)`

**Comment by JsGjKJzi** [comment]
Ok I haven't dug into testing yet so it'll take me a moment to learn the ropes. I also need to think about the possible cases. Maybe it's simple, but I just want to think it through carefully. Hope to have a test added on Monday.

**Comment by JsGjKJzi** [comment]
Actually, it looks like a test case already exists. At first glance it seems to cover this scenario. I'll play with it and try to understand why the test case doesn't seem to jive with actual behavior.
https://github.com/flutter/flutter/blob/8135a4bedac87a8fc61318f1848f7acc9a511964/packages/flutter/test/widgets/interactive_viewer_test.dart#L1807

<details><summary>Test Code</summary>

```dart
    testWidgets('trackpadScrollCausesScale', (WidgetTester tester) async {
      final TransformationController transformationController = TransformationController();
      const double boundaryMargin = 50.0;
      await tester.pumpWidget(
        MaterialApp(
          home: Scaffold(
            body: Center(
              child: InteractiveViewer(
                boundaryMargin: const EdgeInsets.all(boundaryMargin),
                transformationController: transformationController,
                trackpadScrollCausesScale: true,
                child: const SizedBox(width: 200.0, height: 200.0),
              ),
            ),
          ),
        ),
      );

      expect(transformationController.value.getMaxScaleOnAxis(), 1.0);

      // Send a vertical scroll.
      final TestPointer pointer = TestPointer(1, PointerDeviceKind.trackpad);
      final Offset center = tester.getCenter(find.byType(SizedBox));
      await tester.sendEventToBinding(pointer.panZoomStart(center));
      await tester.pump();
      expect(transformationController.value.getMaxScaleOnAxis(), 1.0);
      await tester.sendEventToBinding(pointer.panZoomUpdate(center, pan: const Offset(0, -81)));
      await tester.pump();
      expect(transformationController.value.getMaxScaleOnAxis(), moreOrLessEquals(1.499302500056767));

      // Send a horizontal scroll (should have no effect).
      await tester.sendEventToBinding(pointer.panZoomUpdate(center, pan: const Offset(81, -81)));
      await tester.pump();
      expect(transformationController.value.getMaxScaleOnAxis(), moreOrLessEquals(1.499302500056767));
    });
```

</details>

**Comment by moffatman** [comment]
The current test works for "proper trackpad events" which use the full PointerPanZoom sequence. But on web, we don't have those events (not enough info provided by web APIs). Instead we have PointerScrollEvents with `kind = PointerDeviceKind.trackpad`. And that isn't covered in the test (need to use `pointer.scroll` to create that event instead of `pointer.panZoomStart, etc`)

**Review by justinmc — COMMENTED** [review]
+1 to what @moffatman said.  Otherwise this looks good, thanks for catching this!  I'm happy to rereview when you have a test.

**Comment by JsGjKJzi** [comment]
Ok, I've added a test case to cover this event specifically. I verified locally that the test does not pass without the patch, and does pass with the patch, and the outcome of other test does not change. I'm not sure if there's a way to do that here to verify that the test case.

@justinmc Could you please review as appropriate when you have a moment? Thanks in advance.

@moffatman Thank you for pointing me in the right direction on the test!

File: packages/flutter/test/widgets/interactive_viewer_test.dart [hunk 1]
```diff
@@ -1840,6 +1840,55 @@ void main() {
...
+      // center of the InteractiveViewer
+      Vector3 translation = transformationController.value.getTranslation();
+      expect(translation.x, moreOrLessEquals(-99.37155332430822));
+      expect(translation.y, moreOrLessEquals(-99.37155332430822));
+
+      // Send a horizontal scroll.
+      scrollAmnt = const Offset(-138, 0);
+      await tester.sendEventToBinding(pointer.scroll(scrollAmnt));
+      await tester.pump();
+      expect(transformationController.value.getMaxScaleOnAxis(), moreOrLessEquals(1.9937155332430823));
```
Comment (justinmc) [inline packages/flutter/test/widgets/interactive_viewer_test.dart:1884]: Nit: Maybe add a comment saying that it didn't change the scale either?

**Review by justinmc — APPROVED** [review]
LGTM, thanks for checking that this test fails on master.

File: packages/flutter/test/widgets/interactive_viewer_test.dart [hunk 2]
```diff
@@ -1840,6 +1840,55 @@ void main() {
...
+      final Offset center = tester.getCenter(find.byType(SizedBox));
+      Offset scrollAmnt = const Offset(0, -138.0);
+      await tester.sendEventToBinding(pointer.hover(center));
+      await tester.pump();
+      expect(transformationController.value.getMaxScaleOnAxis(), 1.0);
+      await tester.sendEventToBinding(pointer.scroll(scrollAmnt));
+      await tester.pump();
+      expect(transformationController.value.getMaxScaleOnAxis(), moreOrLessEquals(1.9937155332430823));
+
+      // Scroll should not have translated the box such that the box is still at the
```
[the remark this answers is not in the document]
Comment (JsGjKJzi) [reply packages/flutter/test/widgets/interactive_viewer_test.dart]: Agreed. Thank you for the suggestion.

File: packages/flutter/test/widgets/interactive_viewer_test.dart [hunk 3]
```diff
@@ -1840,6 +1840,55 @@ void main() {
...
+      // center of the InteractiveViewer
```
[the remark this answers is not in the document]
Comment (JsGjKJzi) [reply packages/flutter/test/widgets/interactive_viewer_test.dart]: Re-worded this and included a period.

File: packages/flutter/test/widgets/interactive_viewer_test.dart (hunk 1 above)
Comment (JsGjKJzi) [reply packages/flutter/test/widgets/interactive_viewer_test.dart:1886]: Seems reasonable. Added to comments. Thanks.

**Comment by JsGjKJzi** [comment]
trying to link issue... fixes #127111

**Comment by justinmc** [comment]
@moffatman Could you give a secondary review?

**Review by moffatman — APPROVED** [review]
Looks good

**Comment by justinmc** [comment]
Thanks!

**Pull request merged**